ipn/ipnauth/ipnauth_unix_creds.go

```diff
@@ -18,12 +18,12 @@ import (
 func GetConnIdentity(_ logger.Logf, c net.Conn) (ci *ConnIdentity, err error) {
 	ci = &ConnIdentity{conn: c, notWindows: true}
 	_, ci.isUnixSock = c.(*net.UnixConn)
-	if ci.creds, err = peercred.Get(c); ci.creds != nil {
+	if ci.creds, err = peercred.Get(c); err == nil {
```

Comment:
careful! (because I apparently wasn't) .... this still assigns a nil pointer (making a non-nil interface!) to ci.creds when err != nil here. You don't crash now, but you still leave a dangerous value that'll crash later in `ci.creds`.

You should do like:

      if creds, err := peercred.Get(c); err == nil {
             ci.creds = creds // avoid pointer in ci.creds interface
             ci.pid, _ = creds.PID()